tests/e2e/spec.cy.ts

```diff
@@ -82,14 +82,18 @@ describe("General processes when logged in", () => {
       cy.contains(".account-backend", "comchain").parents(".account").click()
     })
     cy.topUpButton().should("be.visible")
-    cy.topUpButton().click()
-    if (Cypress.env("screenshot")) cy.takeScreenshot("top-up")
-
-    // top-up empty or invalid amount
-    cy.amountInput().should("be.visible")
-    cy.amountInput().type("test invalid amount")
-    cy.topUpNextButton().should("be.disabled")
+    cy.transactionList().should("be.visible")
+    cy.get(".transaction-list").then(($transactionList) => {
+      cy.topUpButton().click()
+      if ($transactionList.find(".unpaid-topup").length) {
+        cy.unpaidTopupModal()
+      } else {
+        // top-up empty or invalid amount
+        cy.topupModal()
```

Comment:
We should probably test both sides in a consistent scenario and not depend on the data we find on the other side (that should be in a predefined state).